### ISSUE TITLE
new netdata documentation - help needed

### ISSUE THREAD

**Issue by ktsaou** [firehol/netdata#1518]
Yes, I know: documentation is a mess.

Well, I think there should be documentation in the following structure.

One way to do it, is to open a new github repo to put all documentation there and then replace the wiki of this repo with the new one. For this to work, this has to be quick (otherwise the 2 wikis will get out of sync pretty quickly).

If you are willing to help me out, please join.

---

# netdata documentation

1. **Introduction**
    - What it is
    - How it works
    - Quick Start Guide

<br/>

2. **Install netdata**
   - General installation instructions for any system
   - Install on Debian/Ubuntu Linux
   - Install on Centos/Fedora/Redhat Enterprise Linux
   - Install on Arch Linux
   - Install on Gentoo Linux
   - Install on Alpine Linux
   - Install on Raspberry Pi
   - Install on FreeBSD
   - Install on MacOS
   - Install on Amazon Machine Image (AMI)
   - Install on Ubuntu on Windows 10
   - Install in a container
	   - Install in docker containers
	   - Install in LXC containers
   - Updating netdata to latest version

<br/>

3. **Configuring netdata**
	- netdata Configuration Guide
	- Tuning CPU consumption (for weak IoT devices)
	- Tuning Memory usage
	- Tuning for high performance (public netdata on the internet)

<br/>

4. **Running netdata**
   - Command Line Options Reference
   - Log Files
   - Tracing Options

<br/>

5. **Data Collection**
    - Monitor Processes, Users & User Groups
	- Monitor Disks
	- Monitor Containers
		- Monitor docker containers
		- Monitor lxc/lxd containers
		- Monitor ephemeral containers
	- Monitor Network Latency (fping)
	- Monitor SNMP Devices
	- Monitor Hardware Sensors
	- Monitor apache Web Servers
	- Monitor nginx Web Servers
	- Monitor tomcat Web Servers
	- Monitor PHP-FPM Applications
	- Monitor mysql/mariadb Databases
	- Monitor postgres Databases
	- Monitor redis Databases
	- Monitor memcached Databases
	- Monitor ISC bind Name Servers
	- Monitor Squid Proxy Servers
	- Monitor OpenSIPs Servers
	- Monitor Mail Servers (postfix, exit, dovecot)
	- Monitor Access Points (hostapd)
	- Monitor UPS (nut, apc)
	- Monitor Fail2Ban Servers
	- Monitor ISC dhcpd Servers
	- Monitor RetroShare (social sharing)
	- Monitor FreeRadius (AAA)
	- Monitor OpenVPN
 
<br/>

6. **my-netdata Menu - netdata Registry**
    - How it works
    - Run your own registry
 
<br/>

7. **Data Archiving - Backends**
   - Introduction - How it works
   - graphite backend
   - opentsdb backend
   - prometheus backend
   - viewing archived data with Grafana

<br/>

8. **Health Monitoring - Alarms**
	- Introduction - How it works
	- Health Monitoring Configuration Guide
		- Health Configuration Files
		- Writing New Alarms
	- Alarm Notifications
		- email notifications
		- pagerduty notifications
		- pushbullet notifications
		- pushover notifications
		- slack.com notifications
		- telegram notifications
		- twilio notifications
	- Troubleshooting Alarms

<br/>

9. **Security Considerations**
	- Securing netdata
		- User authentication via nginx
		- User authentication via apache
		- User authentication via lighttpd
		- User authentication via caddy

<br/>

10. **How to write a Data Collection plugin**
    - Understanding Charts and Metrics
    - Data Collection API Reference
	- Internal Plugins API (C)
	- External Plugins API (any language)
	- charts.d.plugin API (bash)
	- node.d.plugin API (node.js)
	- python.d.plugin API (python)

<br/>

11. **Dashboard Web API**
    - Overview
    - Web API Reference
    - Creating Custom Dashboards
    - Generating Badges

**Comment by diizzyy** [comment]
FreeBSD will more than likely have netdata in ports (package repo) once there's a release.
It's also listed in Wanted Ports. https://wiki.freebsd.org/WantedPorts

**Comment by simonnagl** [comment]
This replaces #378.

The table of contents you posted above looks good. Some thoughts of the old thread:

1. Netdata should get a manage describing the absolute minimum needed. (Mostly where to get further information)
2. For new contributors the header files should be documented much better. I suggest using doxygen. If you like I can focus on the doxygen documentation but I will need your help there.

**Comment by glensc** [comment]
@ktsaou just reminder, if deciding to create new repo, wiki has git url so you can clone to keep the history from wiki's git:

```
git clone https://github.com/firehol/netdata.wiki.git
git remote add newrepo https://github.com/firehol/docs.git
```

actually same applies if deciding to merge with netdata repo, just need to use `--allow-unrelated-histories` option to `git merge`, and perhaps even `git filter-branch` to mv to subdir files

**Comment by ktsaou** [comment]
thanks! I am already keeping a backup of the wiki every few minutes and I am receiving alarms when pages are edited (better be safe than sorry...)

**Comment by kmonsoor** [comment]
@ktsaou though I'm interested to contribute in orchestration of the documentation, I am not sure where to start. I've also browsed other related issues, but get it.  Maybe many others may feel the same. 
Can you please elaborate on how to get started in this.

**Comment by ktsaou** [comment]
nice! it would be perfect if you could help.

Well... I am still trying to find out how to help users more efficiently. Recently I discovered this fantastic draw.io feature that allows us to have an interactive infographic. I have it live at https://my-netdata.io/infographic.html

I would really love something like this, as the main menu to documentation. So users will just click on the icon of the feature or the module they want to know more about, and they should get to a page that has all the information about it.

I have already linked most icons, however, the documentation is not good enough for such occasional reading. It needs quite some work to be organized so that duplicated content will be avoided and provide enough introductory info for the occasional reader.

The draw.io diagram itself will be merged with PR #2121 so it will be in the repo too.

Have a look at it, and let me know of your thoughts...

**Comment by ktsaou** [comment]
replaced with #2638
